azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/ContainerService.java

```diff
@@ -0,0 +1,263 @@
+     */
+    ContainerServiceWindowsProfile windowsProfile();
+
+    /**
+     * @return the properties for the Linux VMs.
+     */
+    ContainerServiceLinuxProfile linuxProfile();
+
+    /**
+     * @return the properties for the diagnostic agent.
+     */
+    ContainerServiceDiagnosticsProfile diagnosticsProfile();
+
+    // Fluent interfaces
+
+    /**
+     * Container interface for all the definitions related to a container service.
+     */
+    interface Definition extends
+            ContainerService.DefinitionStages.Blank,
+            ContainerService.DefinitionStages.WithGroup,
+            ContainerService.DefinitionStages.WithMasterProfile,
+            ContainerService.DefinitionStages.WithLinuxProfile,
+            ContainerService.DefinitionStages.WithLinuxProfileRootUsername,
+            ContainerService.DefinitionStages.WithLinuxProfileSshKey,
+            ContainerService.DefinitionStages.DefineAgentPoolProfiles,
+            ContainerService.DefinitionStages.WithCreate {
+    }
+
+    /**
+     * Grouping of container service definition stages.
+     */
+    interface DefinitionStages {
+        /**
+         * The first stage of a container service definition.
+         */
+        interface Blank extends
+                GroupableResource.DefinitionWithRegion<WithGroup> {
+        }
+
+        /**
+         * The stage of the container service definition allowing to specify the resource group.
+         */
+        interface WithGroup extends
+                GroupableResource.DefinitionStages.WithGroup<WithMasterProfile> {
+        }
+
+        /**
+         * The stage of the container service definition allowing to specify orchestration type.
+         */
+        interface WithOrchestratorProfile {
+            /**
+             * Specifies the Swarm orchestration type for the container service.
+             * @return the next stage of the definition
+             */
+            WithCreate withSwarmOrchestration();
+
+            /**
+             * Specifies the DCOS orchestration type for the container service.
+             * @return the next stage of the definition
+             */
+            WithCreate withDCOSOrchestration();
+
+            /**
+             * Specifies the Kubernetes orchestration type for the container service.
+             * @return the next stage of the definition
+             */
+            WithCreate withKubernetesOrchestration();
+        }
+
+        /**
+         * The stage of the container service definition allowing to specify the master profile.
+         */
+        interface WithMasterProfile {
+            /**
+             * The properties for master agents.
+             * @param count master profile count (1, 3, 5)
+             * @param dnsLabel the dns prefix
+             * @return the next stage of the definition
+             */
+            WithLinuxProfile withMasterProfile(ContainerServiceMasterProfileCount count, String dnsLabel);

```

Comment:
this feels a bit 'old style auto-gen'd SDK'-ish. Do we need to expose the `master profile` as a concept that the user needs to understand? Isn't the count just the number of master nodes and the dnsLabel a prefix for the DNS of the master cluster, or something like that?  If so, how about splitting these into a sequence of two withers, like:
`.withMasterNodeCount()`
`.withMasterDnsLabel()`
?